lib/puppet/provider/mount/parsed.rb

```diff
@@ -178,7 +178,21 @@ def to_line(result)
       end
     end
   else
-    record_line self.name, :fields => @fields, :separator => /\s+/, :joiner => "\t", :optional => optional_fields
+    record_line self.name, :fields => @fields, :separator => /\s+/, :joiner => "\t", :optional => optional_fields, :block_eval => :instance do
```

Comment:
BTW: You can also pass `:pre_gen => proc {} ` here as in `provider/ssh_authorized_key/parsed.rb`. I guess just a matter of taste :)